sound/soc/codecs/es8326.c

```diff
@@ -0,0 +1,854 @@
+		{256,8000,2048000,0x60,0x00,0x11,0x35,0x0A,0x1B,0x1F,0x7F}, 
+		{256,16000,4096000,0x20,0x00,0x01,0x35,0x0A,0x1B,0x1F,0x3F}, 
+
+		{256,44100,11289600,0x00,0x00,0x10,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{256,48000,12288000,0x00,0x00,0x30,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{288,8000,2304000,0x20,0x00,0x01,0x35,0x0A,0x1B,0x23,0x47}, 
+		{384,8000,3072000,0x60,0x02,0x05,0x75,0x0A,0x1B,0x1F,0x7F}, 
+		{384,16000,6144000,0x20,0x02,0x03,0x35,0x0A,0x1B,0x1F,0x3F}, 
+		{384,48000,18432000,0x00,0x02,0x01,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{400,48000,19200000,0x09,0x04,0x0f,0x6d,0x3a,0x0A,0x4F,0x1F}, 
+		{500,48000,24000000,0x18,0x04,0x1F,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{512,8000,4096000,0x60,0x00,0x01,0x35,0x0A,0x1B,0x1F,0x7F}, 
+		{512,16000,8192000,0x20,0x00,0x10,0x35,0x0A,0x1B,0x1F,0x3F}, 
+
+		{512,44100,22579200,0x00,0x00,0x00,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{512,48000,24576000,0x00,0x00,0x00,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{768,8000,6144000,0x60,0x02,0x11,0x35,0x0A,0x1B,0x1F,0x7F}, 
+		{768,16000,12288000,0x20,0x02,0x01,0x35,0x0A,0x1B,0x1F,0x3F}, 
+		{800,48000,38400000,0x00,0x18,0x13,0x2D,0x0A,0x0A,0x1F,0x1F}, 
+		{1024,8000,8192000,0x60,0x00,0x10,0x35,0x0A,0x1B,0x1F,0x7F}, 
+		{1024,16000,16384000,0x20,0x00,0x00,0x35,0x0A,0x1B,0x1F,0x3F}, 
+		{1152,16000,18432000,0x20,0x08,0x11,0x35,0x0A,0x1B,0x1F,0x3F}, 
+		{1536,8000,12288000,0x60,0x02,0x01,0x35,0x0A,0x1B,0x1F,0x7F}, 
+
+		{1536,16000,24576000,0x20,0x02,0x10,0x35,0x0A,0x1B,0x1F,0x3F}, 
+		{1625,8000,13000000,0x0C,0x18,0x1F,0x2D,0x0A,0x0A,0x27,0x27}, 
+		{1625,16000,26000000,0x0C,0x18,0x1F,0x2D,0x0A,0x0A,0x27,0x27}, 
+		{2048,8000,16384000,0x60,0x00,0x00,0x35,0x0A,0x1B,0x1F,0x7F}, 
+		{2304,8000,18432000,0x40,0x02,0x10,0x35,0x0A,0x1B,0x1F,0x5F}, 
+		{3072,8000,24576000,0x60,0x02,0x10,0x35,0x0A,0x1B,0x1F,0x7F}, 
+		{3250,8000,26000000,0x0C,0x18,0x0F,0x2D,0x0A,0x0A,0x27,0x27}, 
+
+};
+
+static inline int get_coeff(int mclk, int rate)
+{
+		int i;
+		rate=ES8326_LRCK;
+		for (i = 0; i < ARRAY_SIZE(coeff_div); i++) {
+				if (coeff_div[i].rate == rate && coeff_div[i].mclk == mclk)
+						return i;
+		}
+
+		return -EINVAL;
+}
+
+static unsigned int rates_12288[] = {
+		8000, 12000, 16000, 24000, 24000, 32000, 48000, 96000,
+};
+
+static struct snd_pcm_hw_constraint_list constraints_12288 = {
+		.count = ARRAY_SIZE(rates_12288),
+		.list = rates_12288,
+};
+
+static unsigned int rates_112896[] = {
+	8000, 11025, 22050, 44100,
+};
+
+static struct snd_pcm_hw_constraint_list constraints_112896 = {
+		.count = ARRAY_SIZE(rates_112896),
+		.list = rates_112896,
+};
+
+static unsigned int rates_12[] = {
+		8000, 11025, 12000, 16000, 22050, 24000, 32000, 44100, 48000,
+		48000, 88235, 96000,
+};
+
+static struct snd_pcm_hw_constraint_list constraints_12 = {
+		.count = ARRAY_SIZE(rates_12),
+		.list = rates_12,
+};
+
+/*
+ * Note that this should be called from init rather than from hw_params.
+ */
+static int es8326_set_dai_sysclk(struct snd_soc_dai *codec_dai,
```

Comment:
Can I get rid of the constraints?  I tried to remove them but then I see a kernel panic

Reply:
that's your code, I don't know what this does.